### DISCUSSION TITLE
why create different tag for k3s.io kubernetes stage repo.

### DISCUSSION THREAD

**Discussion by xiaods** [k3s-io/k3s#5224] [Q&A] [answered]
https://github.com/k3s-io/kubernetes/tags

such as for tag  [v1.23.4-k3s1](https://github.com/k3s-io/kubernetes/releases/tag/v1.23.4-k3s1) based on commit tag: [46139e9](https://github.com/k3s-io/kubernetes/commit/46139e94e03f2a5ebc81fc11aa7d441a33d7de93)

as same with  tag staging/src/k8s.io/sample-controller/v1.23.4-k3s1  on same commit tag:  [46139e9](https://github.com/k3s-io/kubernetes/commit/46139e94e03f2a5ebc81fc11aa7d441a33d7de93)

can  anyone clean my question?

**Comment by brandond** [comment, marked as answer]
Upstream kubernetes moves a bunch of stuff out to individual packages when they tag a new release; check the docs at https://github.com/kubernetes/kubernetes/blob/master/staging/README.md

Due to how go modules work, in order to replace all of these with the code in our fork, we have to tag each of the individual module versions in our fork.
